Binary install script for Mac OSX and Linux

Gable-github opened the pull request:
### Description:
Add a install shell script for easy binary installation of Opengrep.

### Includes:
- install.sh: Downloads and installs the correct prebuilt binary for Linux and macOS (x86_64 and arm64/aarch64), with automatic detection.

- Versioned installs in ~/.opengrep/cli/$VERSION and a symlink to latest.

- Attempts to symlink to ~/.local/bin/opengrep for user-wide CLI access (falls back to a hint for the user to do it manually)


- Handles argument parsing:

    `-v` install specific version
    
    `-l` list available versions
    
    `-h` help/usage

### Why?
Simplifies Opengrep setup for end users (no manual downloads or system-wide installs required), no need for sudo permissions for script to run successfully.

### Docs:
Updated README and install instructions for end users.

### Issue:
Resolves Enchancement Issue #82

JonZeolla commented on install.sh:
```diff
@@ -0,0 +1,191 @@
+# Function to get latest release version
+get_latest_version() {
+    command -v curl >/dev/null 2>&1 || { echo >&2 "Required tool curl could not be found. Aborting."; exit 1; }
+    curl -s https://api.github.com/repos/opengrep/opengrep/releases | grep '"tag_name":' | head -1 | sed -E 's/.*"([^"]+)".*/\1/'
```
I confirmed this works:
```suggestion
    curl -s https://api.github.com/repos/opengrep/opengrep/releases/latest | grep '"tag_name":' | head -1 | sed -E 's/.*"([^"]+)".*/\1/'
```

dimitris-m commented on README.md:
```diff
@@ -29,8 +29,26 @@ Apex · Bash · C · C++ · C# · Clojure · Dart · Dockerfile · Elixir · HTM
+Which will install the latest version of Opengrep.
+
+You can also install manually:
+* Binaries available in the latest alpha [release](https://github.com/opengrep/opengrep/releases).
```
The releases are not alpha any more.

dimitris-m commented on README.md:
```diff
@@ -29,8 +29,26 @@ Apex · Bash · C · C++ · C# · Clojure · Dart · Dockerfile · Elixir · HTM
 
-Binaries available in the latest alpha [release](https://github.com/opengrep/opengrep/releases).
- 
+You can install Opengrep via our official install script.
```
"You can install Opengrep using our official install script."

dimitris-m commented on install.sh:
```diff
@@ -0,0 +1,191 @@
+        echo "Successfully installed Opengrep binary to ${INST}/opengrep"
+        
+        ln -s "${INST}" "${LATEST}" || exit 1
+        echo "with a symlink from ${LATEST}/opengrep"
```
```
echo "Created symlink to ${LATEST}/opengrep"
```

dimitris-m commented on install.sh:
```diff
@@ -0,0 +1,191 @@
+        echo
+        echo "Successfully installed Opengrep binary to ${INST}/opengrep"
+        
+        ln -s "${INST}" "${LATEST}" || exit 1
```
You might need:

```
rm -f "${LATEST}" || exit 1
```

before creating the new symlink.

Gable-github replied:
hi, thanks for the detailed review. I'll be updating the script such that the previous symlink to latest is (1) overriden and (2) only if the most recently installed version is a higher version number.

so latest points to the latest version released by the team instead of the most recent binary installed by the user.

Gable-github replied:
i'll commit it in duckdb way first, as ive tested that and it works.

dimitris-m replied:
I think duckdb way.

The default executable in the path should be the latest installed version.

Gable-github replied:
Okay, it is implemented. Thanks for the guidance!

dimitris-m commented on install.sh:
```diff
@@ -0,0 +1,191 @@
+        echo
+        echo "To launch Opengrep now, type"
+        echo "opengrep"
+    else
+        echo
+        echo "To launch Opengrep now, type"
+        echo "${LATEST}/opengrep"
+    fi
```
When can this happen?

Gable-github replied:
Hi, the flawed logic went over my head. made the changes, it should work now. Thanks!!

dimitris-m reviewed (changes requested):
Left some comments, the most important is to ensure that installation 2 times in a row succeeds.

Also, for merging I will squash the commits to maintain a more or less linear looking history, once you address any issues.

Gable-github commented:
Hi @dimitris-m, thank you for the detailed review. I made the requested changes and cleaned up the logic. 

Note: 
- `latest` symlink at `/.local/bin` will point to the `latest` symlink in `/.opengrep/cli`
- `latest` symlink in `/.opengrep/cli` will update to point to the most recently installed version of opengrep binary by the user, following [the duckdb install script](https://install.duckdb.org/)
- cleaned up the logic for `echo Hint...` for user to manually update the PATH variable. This logic only fires off when the script dosent have permission to do it on their behalf. It was firing off erroneously in the previous commit because of this line `[ ! -f "${LOCALBIN}/opengrep" ]`. I have nested this logic.

dimitris-m reviewed (approved):
Looks good, thanks! 

Merging now.
